plugins/woocommerce/src/Blocks/BlockTypes/ProductFilters.php

```diff
@@ -70,6 +70,38 @@ protected function enqueue_data( array $attributes = [] ) {
 		$this->asset_data_registry->add( 'isWidgetEditor', 'widgets.php' === $pagenow || 'customize.php' === $pagenow, true );
 	}
 
+	/**
+	 * Check if the collection data is empty.
+	 *
+	 * @param mixed $attributes - Block attributes.
+	 * @return bool - Whether the collection data is empty.
+	 */
+	private function collection_data_is_empty( $attributes ) {
+		if ( empty( $this->current_response ) ) {
+			return true;
+		}
+
+		$filter_type = $attributes['filterType'];
+
+		if ( 'attribute-filter' === $filter_type ) {
+			return empty( $this->current_response['attribute_counts'] );
+		}
+
+		if ( 'rating-filter' === $filter_type ) {
+			return empty( $this->current_response['rating_counts'] );
+		}
+
+		if ( 'price-filter' === $filter_type ) {
+			return empty( $this->current_response['price_range'] ) || ( $this->current_response['price_range']['min_price'] === $this->current_response['price_range']['max_price'] );
+		}
+
+		if ( 'stock-filter' === $filter_type ) {
+			return empty( $this->current_response['stock_status_counts'] );
+		}
+
+		return false;
+	}
```

Comment:
We haven't handled the active filter here yet.

Reply:
Active Filters block can be challenged here as we're no longer aggregate the filter data. So it may need a different way to control the visibility.

Reply:
@dinhtungdu I duplicated logic extracting the filter query params here instead, if they're empty then I don't display it.

Let me know what you think. I considered DRY of the code for the params, but ultimately felt there was no sensible place to put it and opted instead for a small amount of duplication.

https://github.com/woocommerce/woocommerce/pull/43688/commits/a6761325432ea782dab9d50c8cc0e48340b8f51b